packages/ckeditor5-dev-release-tools/tests/tasks/preparerepository.js

```diff
@@ -0,0 +1,321 @@
+				stubs.fs.lstat.withArgs( 'current/working/dir/packages/textFile.txt' ).resolves( stubs.lstat.isNotDir );
+				stubs.fs.existsSync.withArgs( 'current/working/dir/packages/ckeditor5-core/package.json' ).resolves( true );
+				stubs.fs.existsSync.withArgs( 'current/working/dir/packages/ckeditor5-utils/package.json' ).resolves( true );
+
+				await prepareRepository( options );
+
+				expect( stubs.fs.copy.callCount ).to.equal( 2 );
+
+				expect( stubs.fs.copy.getCall( 0 ).args.length ).to.equal( 2 );
+				expect( stubs.fs.copy.getCall( 0 ).args[ 0 ] ).to.equal( 'current/working/dir/packages/ckeditor5-core' );
+				expect( stubs.fs.copy.getCall( 0 ).args[ 1 ] ).to.equal( 'current/working/dir/release/ckeditor5-core' );
+
+				expect( stubs.fs.copy.getCall( 1 ).args.length ).to.equal( 2 );
+				expect( stubs.fs.copy.getCall( 1 ).args[ 0 ] ).to.equal( 'current/working/dir/packages/ckeditor5-utils' );
+				expect( stubs.fs.copy.getCall( 1 ).args[ 1 ] ).to.equal( 'current/working/dir/release/ckeditor5-utils' );
+			} );
+
+			it( 'should not copy directories that do not have the "package.json" file', async () => {
+				options.packagesDirectory = 'packages';
+
+				stubs.fs.lstat.withArgs( 'current/working/dir/packages/ckeditor5-core' ).resolves( stubs.lstat.isDir );
+				stubs.fs.lstat.withArgs( 'current/working/dir/packages/ckeditor5-utils' ).resolves( stubs.lstat.isDir );
+				stubs.fs.existsSync.withArgs( 'current/working/dir/packages/ckeditor5-core/package.json' ).resolves( true );
+				stubs.fs.existsSync.withArgs( 'current/working/dir/packages/ckeditor5-utils/package.json' ).resolves( false );
+
+				await prepareRepository( options );
+
+				expect( stubs.fs.copy.callCount ).to.equal( 1 );
+
+				expect( stubs.fs.copy.getCall( 0 ).args.length ).to.equal( 2 );
+				expect( stubs.fs.copy.getCall( 0 ).args[ 0 ] ).to.equal( 'current/working/dir/packages/ckeditor5-core' );
+				expect( stubs.fs.copy.getCall( 0 ).args[ 1 ] ).to.equal( 'current/working/dir/release/ckeditor5-core' );
+			} );
+
+			it( 'should copy only the specified packages if the "packagesToCopy" option is provided', async () => {
+				options.packagesDirectory = 'packages';
+				options.packagesToCopy = [ 'ckeditor5-core' ];
+
+				stubs.fs.lstat.withArgs( 'current/working/dir/packages/ckeditor5-core' ).resolves( stubs.lstat.isDir );
+				stubs.fs.existsSync.withArgs( 'current/working/dir/packages/ckeditor5-core/package.json' ).resolves( true );
+
+				await prepareRepository( options );
+
+				expect( stubs.fs.copy.callCount ).to.equal( 1 );
+
+				expect( stubs.fs.copy.getCall( 0 ).args.length ).to.equal( 2 );
+				expect( stubs.fs.copy.getCall( 0 ).args[ 0 ] ).to.equal( 'current/working/dir/packages/ckeditor5-core' );
+				expect( stubs.fs.copy.getCall( 0 ).args[ 1 ] ).to.equal( 'current/working/dir/release/ckeditor5-core' );
+			} );
+
+			it( 'should allow copying nested packages via the "packagesToCopy" option', async () => {
```

Comment:
Still determining if we need such a test. I cannot imagine a real use case. Do you?